Address additional API feedback for Web Extensions.

xeenon opened the pull request:
#### 9d9281b1550693b34eae187e13642ae1329016b1
<pre>
Address additional API feedback for Web Extensions.
<a href="https://webkit.org/b/278040">https://webkit.org/b/278040</a>
<a href="https://rdar.apple.com/problem/133783321">rdar://problem/133783321</a>

Reviewed by Jeff Miller.

Address various API review comments with tweaks to various property names, header docs, and some classes.

* Source/WebCore/en.lproj/Localizable.strings: Updated with update-webkit-localizable-strings.
* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKit/Modules/iOS_Private.modulemap:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.mm:
(-[WKWebExtension hasPersistentBackgroundContent]):
(-[WKWebExtension _hasServiceWorkerBackgroundContent]):
(-[WKWebExtension _hasModularBackgroundContent]):
(-[WKWebExtension backgroundContentIsPersistent]): Deleted.
(-[WKWebExtension _backgroundContentIsServiceWorker]): Deleted.
(-[WKWebExtension _backgroundContentUsesModules]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionAction.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommand.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommandPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.mm:
(-[WKWebExtensionContext hasRequestedOptionalAccessToAllHosts]):
(-[WKWebExtensionContext setHasRequestedOptionalAccessToAllHosts:]):
(-[WKWebExtensionContext hasAccessToPrivateData]):
(-[WKWebExtensionContext setHasAccessToPrivateData:]):
(toImpl):
(-[WKWebExtensionContext requestedOptionalAccessToAllHosts]): Deleted.
(-[WKWebExtensionContext setRequestedOptionalAccessToAllHosts:]): Deleted.
(-[WKWebExtensionContext hasAccessInPrivateBrowsing]): Deleted.
(-[WKWebExtensionContext setHasAccessInPrivateBrowsing:]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfigurationPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.mm:
(-[WKWebExtensionMessagePort sendMessage:completionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTab.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTabConfiguration.h: Renamed from WKWebExtensionTabCreationOptions.h.
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTabConfiguration.mm: Renamed from WKWebExtensionTabCreationOptions.mm.
(-[WKWebExtensionTabConfiguration _init]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTabConfigurationInternal.h: Renamed from WKWebExtensionTabCreationOptionsInternal.h.
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindow.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindowConfiguration.h: Renamed from WKWebExtensionWindowCreationOptions.h.
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindowConfiguration.mm: Renamed from WKWebExtensionWindowCreationOptions.mm.
(-[WKWebExtensionWindowConfiguration _init]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindowConfigurationInternal.h: Renamed from WKWebExtensionWindowCreationOptionsInternal.h.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionTab.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionTabCreationOptions.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionWindowCreationOptions.h:
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIExtensionCocoa.mm:
(WebKit::WebExtensionContext::extensionIsAllowedIncognitoAccess):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIRuntimeCocoa.mm:
(WebKit::WebExtensionContext::runtimeOpenOptionsPage):
(WebKit::WebExtensionContext::runtimeSendNativeMessage):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPITabsCocoa.mm:
(WebKit::WebExtensionContext::tabsCreate):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIWindowsCocoa.mm:
(WebKit::WebExtensionContext::windowsCreate):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm:
(WebKit::WebExtension::populateBackgroundPropertiesIfNeeded):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::setHasAccessToPrivateData):
(WebKit::WebExtensionContext::canOpenNewWindow const):
(WebKit::WebExtensionContext::userGesturePerformed):
(WebKit::WebExtensionContext::websiteDataStore const):
(WebKit::WebExtensionContext::setHasAccessInPrivateBrowsing): Deleted.
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionControllerCocoa.mm:
(WebKit::WebExtensionController::addUserContentController):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionTabCocoa.mm:
(WebKit::WebExtensionTab::WebExtensionTab):
(WebKit::WebExtensionTab::parameters const):
(WebKit::WebExtensionTab::changedParameters const):
(WebKit::WebExtensionTab::matches const):
(WebKit::WebExtensionTab::extensionHasAccess const):
(WebKit::WebExtensionTab::setReaderModeShowing):
(WebKit::WebExtensionTab::isReaderModeShowing const):
(WebKit::WebExtensionTab::isPlayingAudio const):
(WebKit::WebExtensionTab::detectWebpageLocale):
(WebKit::WebExtensionTab::captureVisibleWebpage):
(WebKit::WebExtensionTab::duplicate):
(WebKit::WebExtensionTab::shouldGrantPermissionsOnUserGesture const):
(WebKit::WebExtensionTab::isAudible const): Deleted.
(WebKit::WebExtensionTab::shouldGrantTabPermissionsOnUserGesture const): Deleted.
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionWindowCocoa.mm:
(WebKit::WebExtensionWindow::WebExtensionWindow):
(WebKit::WebExtensionWindow::extensionHasAccess const):
(WebKit::WebExtensionWindow::isPrivate const):
* Source/WebKit/UIProcess/Extensions/WebExtension.serialization.in:
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.cpp:
(WebKit::WebExtensionContext::userContentControllers const):
(WebKit::WebExtensionContext::pageListensForEvent const):
(WebKit::WebExtensionContext::processes const):
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
(WebKit::WebExtensionContext::hasAccessToPrivateData const):
(WebKit::WebExtensionContext::hasAccessInPrivateBrowsing const): Deleted.
* Source/WebKit/UIProcess/Extensions/WebExtensionTab.h:
(WebKit::WebExtensionTab::allChangedProperties):
* Source/WebKit/UIProcess/Extensions/WebExtensionWindow.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtension.mm:
(TestWebKitAPI::TEST(WKWebExtension, BackgroundParsing)):
(TestWebKitAPI::TEST(WKWebExtension, BackgroundPreferredEnvironmentParsing)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIAction.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIAction, NavigationOpensInNewTab)):
(TestWebKitAPI::TEST(WKWebExtensionAPIAction, WindowOpenOpensInNewWindow)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPICookies.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPICookies, GetAllCookieStoresWithPrivateAccess)):
(TestWebKitAPI::TEST(WKWebExtensionAPICookies, GetAllIncognitoWithPrivateAccess)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIDeclarativeNetRequest.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIDeclarativeNetRequest, BlockedLoadInPrivateBrowsingTest)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIExtension.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIExtension, InIncognitoContext)):
(TestWebKitAPI::TEST(WKWebExtensionAPIExtension, IsAllowedIncognitoAccess)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIRuntime.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIRuntime, ConnectNative)):
(TestWebKitAPI::TEST(WKWebExtensionAPIRuntime, ConnectNativeWithInvalidMessage)):
(TestWebKitAPI::TEST(WKWebExtensionAPIRuntime, ConnectNativeWithUndefinedMessage)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPITabs.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, Create)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, CreateWithSpecifiedOptions)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, CreateWithRelativeURL)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, Duplicate)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, DuplicateWithOptions)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, QueryWithPrivateAccess)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, UpdatedEventWithPrivateAccess)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIWindows.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, GetAll)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, GetAllWithPrivateAccess)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, Create)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, CreateWithRelativeURL)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, CreateWithRelativeURLs)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, CreateIncognitoWithoutPrivateAccess)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, CreateIncognitoWithPrivateAccess)):
* Tools/TestWebKitAPI/cocoa/TestWebExtensionsDelegate.h:
* Tools/TestWebKitAPI/cocoa/TestWebExtensionsDelegate.mm:
(-[TestWebExtensionsDelegate webExtensionController:openNewWindowUsingConfiguration:forExtensionContext:completionHandler:]):
(-[TestWebExtensionsDelegate webExtensionController:openNewTabUsingConfiguration:forExtensionContext:completionHandler:]):
(-[TestWebExtensionsDelegate webExtensionController:sendMessage:toApplicationWithIdentifier:forExtensionContext:replyHandler:]):
(-[TestWebExtensionsDelegate webExtensionController:openNewWindowWithOptions:forExtensionContext:completionHandler:]): Deleted.
(-[TestWebExtensionsDelegate webExtensionController:openNewTabWithOptions:forExtensionContext:completionHandler:]): Deleted.
(-[TestWebExtensionsDelegate webExtensionController:sendMessage:toApplicationIdentifier:forExtensionContext:replyHandler:]): Deleted.
* Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.h:
* Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.mm:
(-[TestWebExtensionManager initForExtension:extensionControllerConfiguration:]):
(-[TestWebExtensionTab isReaderModeActiveForWebExtensionContext:]):
(-[TestWebExtensionTab setReaderModeActive:forWebExtensionContext:completionHandler:]):
(-[TestWebExtensionTab duplicateForWebExtensionContext:withOptions:completionHandler:]):
(-[TestWebExtensionWindow openNewTabAtIndex:]):
(-[TestWebExtensionWindow isPrivateForWebExtensionContext:]):
(-[TestWebExtensionTab isReaderModeShowingForWebExtensionContext:]): Deleted.
(-[TestWebExtensionTab setReaderModeShowing:forWebExtensionContext:completionHandler:]): Deleted.
(-[TestWebExtensionWindow isUsingPrivateBrowsingForWebExtensionContext:]): Deleted.

Canonical link: <a href="https://commits.webkit.org/282252@main">https://commits.webkit.org/282252@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1a83ad51c4fecf5892dfb5175d61f58d860d92b6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62582 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41937 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15177 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/66566 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13134 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/64702 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/49624 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/13470 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50423 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9057 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65651 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/38975 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54203 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31165 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/35697 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11538 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12062 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57283 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/11854 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/68297 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6528 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/11541 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/57799 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6558 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54250 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/57993 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5449 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9422 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/37737 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/38822 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/39919 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/38566 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jeffmapple commented on Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerDelegate.h:
```diff
@@ -49,52 +49,52 @@ WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA)) WK_S
- the creation based on the provided options and existing windows. Once handled, the app should call the completion handler with the created window
- or `nil` if the creation was declined or failed. If not implemented, the extension can't open new windows.
+ @discussion This method should be implemented by the app to handle requests to open new windows. The app can decide how to handle the
+ process based on the provided configuation and existing windows. Once handled, the app should call the completion handler with the opened window
```
```suggestion
 process based on the provided configuration and existing windows. Once handled, the app should call the completion handler with the opened window
```

jeffmapple commented on Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIWindowsCocoa.mm:
```diff
@@ -55,11 +55,11 @@
-    creationOptions.windowState = toAPI(creationParameters.state.value_or(WebExtensionWindow::State::Normal));
-    creationOptions.focused = creationParameters.focused.value_or(true);
-    creationOptions.usePrivateBrowsing = creationParameters.privateBrowsing.value_or(false);
+    auto *configruation = [[WKWebExtensionWindowConfiguration alloc] _init];
```
```suggestion
    auto *configuration = [[WKWebExtensionWindowConfiguration alloc] _init];
```

jeffmapple commented on Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionTabCocoa.mm:
```diff
@@ -125,14 +125,14 @@
 
         m_respondsToIsReaderModeAvailable ? std::optional(isReaderModeAvailable()) : std::nullopt,
-        m_respondsToIsReaderModeShowing ? std::optional(isReaderModeShowing()) : std::nullopt
+        m_respondsToIsReaderModeActive ? std::optional(isReaderModeShowing()) : std::nullopt
```
Should this be:
```suggestion
        m_respondsToIsReaderModeActive ? std::optional(isReaderModeActive()) : std::nullopt
```

jeffmapple commented on Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionTabCocoa.mm:
```diff
@@ -159,15 +159,15 @@
 
         m_respondsToIsReaderModeAvailable && changedProperties.contains(ChangedProperties::ReaderMode) ? std::optional(isReaderModeAvailable()) : std::nullopt,
-        m_respondsToIsReaderModeShowing && changedProperties.contains(ChangedProperties::ReaderMode) ? std::optional(isReaderModeShowing()) : std::nullopt
+        m_respondsToIsReaderModeActive && changedProperties.contains(ChangedProperties::ReaderMode) ? std::optional(isReaderModeShowing()) : std::nullopt
```
Should this be:
```suggestion
        m_respondsToIsReaderModeActive && changedProperties.contains(ChangedProperties::ReaderMode) ? std::optional(isReaderModeActive()) : std::nullopt
```

jeffmapple commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIWindows.mm:
```diff
@@ -431,14 +431,14 @@ @implementation DummyWebExtensionWindow
     auto originalOpenNewWindow = manager.get().internalDelegate.openNewWindow;
 
-    manager.get().internalDelegate.openNewWindow = ^(WKWebExtensionWindowCreationOptions *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
+    manager.get().internalDelegate.openNewWindow = ^(WKWebExtensionWindowConfiguration *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
```
```suggestion
    manager.get().internalDelegate.openNewWindow = ^(WKWebExtensionWindowConfiguration *configuration, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
```

jeffmapple commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIWindows.mm:
```diff
@@ -507,7 +507,7 @@ @implementation DummyWebExtensionWindow
     auto originalOpenNewWindow = manager.get().internalDelegate.openNewWindow;
 
-    manager.get().internalDelegate.openNewWindow = ^(WKWebExtensionWindowCreationOptions *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
+    manager.get().internalDelegate.openNewWindow = ^(WKWebExtensionWindowConfiguration *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
```
```suggestion
    manager.get().internalDelegate.openNewWindow = ^(WKWebExtensionWindowConfiguration *configuration, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
```

jeffmapple commented on Tools/TestWebKitAPI/cocoa/TestWebExtensionsDelegate.mm:
```diff
@@ -52,7 +52,7 @@ @implementation TestWebExtensionsDelegate
 
 #if PLATFORM(MAC)
-- (void)webExtensionController:(WKWebExtensionController *)controller openNewWindowWithOptions:(WKWebExtensionWindowCreationOptions *)options forExtensionContext:(WKWebExtensionContext *)extensionContext completionHandler:(void (^)(id<WKWebExtensionWindow> newWindow, NSError *error))completionHandler
+- (void)webExtensionController:(WKWebExtensionController *)controller openNewWindowUsingConfiguration:(WKWebExtensionWindowConfiguration *)options forExtensionContext:(WKWebExtensionContext *)extensionContext completionHandler:(void (^)(id<WKWebExtensionWindow> newWindow, NSError *error))completionHandler
```
```suggestion
- (void)webExtensionController:(WKWebExtensionController *)controller openNewWindowUsingConfiguration:(WKWebExtensionWindowConfiguration *)configuration forExtensionContext:(WKWebExtensionContext *)extensionContext completionHandler:(void (^)(id<WKWebExtensionWindow> newWindow, NSError *error))completionHandler
```

jeffmapple commented on Tools/TestWebKitAPI/cocoa/TestWebExtensionsDelegate.mm:
```diff
@@ -61,7 +61,7 @@ - (void)webExtensionController:(WKWebExtensionController *)controller openNewWin
 #endif
 
-- (void)webExtensionController:(WKWebExtensionController *)controller openNewTabWithOptions:(WKWebExtensionTabCreationOptions *)options forExtensionContext:(WKWebExtensionContext *)extensionContext completionHandler:(void (^)(id<WKWebExtensionTab> newTab, NSError *error))completionHandler
+- (void)webExtensionController:(WKWebExtensionController *)controller openNewTabUsingConfiguration:(WKWebExtensionTabConfiguration *)options forExtensionContext:(WKWebExtensionContext *)extensionContext completionHandler:(void (^)(id<WKWebExtensionTab> newTab, NSError *error))completionHandler
```
```suggestion
- (void)webExtensionController:(WKWebExtensionController *)controller openNewTabUsingConfiguration:(WKWebExtensionTabConfiguration *)configuration forExtensionContext:(WKWebExtensionContext *)extensionContext completionHandler:(void (^)(id<WKWebExtensionTab> newTab, NSError *error))completionHandler
```

jeffmapple commented on Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.mm:
```diff
@@ -88,8 +88,8 @@ - (instancetype)initForExtension:(WKWebExtension *)extension extensionController
 
-    _internalDelegate.openNewWindow = ^(WKWebExtensionWindowCreationOptions *options, WKWebExtensionContext *, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
-        auto *newWindow = [weakSelf openNewWindowUsingPrivateBrowsing:options.usePrivateBrowsing];
+    _internalDelegate.openNewWindow = ^(WKWebExtensionWindowConfiguration *options, WKWebExtensionContext *, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
```
```suggestion
    _internalDelegate.openNewWindow = ^(WKWebExtensionWindowConfiguration *configuration, WKWebExtensionContext *, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
```

jeffmapple commented on Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.mm:
```diff
@@ -120,7 +120,7 @@ - (instancetype)initForExtension:(WKWebExtension *)extension extensionController
 #endif // PLATFORM(MAC)
 
-    _internalDelegate.openNewTab = ^(WKWebExtensionTabCreationOptions *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionTab>, NSError *)) {
+    _internalDelegate.openNewTab = ^(WKWebExtensionTabConfiguration *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionTab>, NSError *)) {
```
```suggestion
    _internalDelegate.openNewTab = ^(WKWebExtensionTabConfiguration *configuration, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionTab>, NSError *)) {
```

jeffmapple commented on Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.mm:
```diff
@@ -527,7 +527,7 @@ - (void)setMuted:(BOOL)muted forWebExtensionContext:(WKWebExtensionContext *)con
 }
 
-- (void)duplicateForWebExtensionContext:(WKWebExtensionContext *)context withOptions:(WKWebExtensionTabCreationOptions *)options completionHandler:(void (^)(id<WKWebExtensionTab>, NSError *))completionHandler
+- (void)duplicateForWebExtensionContext:(WKWebExtensionContext *)context withOptions:(WKWebExtensionTabConfiguration *)options completionHandler:(void (^)(id<WKWebExtensionTab>, NSError *))completionHandler
```
```suggestion
- (void)duplicateForWebExtensionContext:(WKWebExtensionContext *)context withConfiguration:(WKWebExtensionTabConfiguration *)configuration completionHandler:(void (^)(id<WKWebExtensionTab>, NSError *))completionHandler
```

jeffmapple commented on Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.mm:
```diff
@@ -681,15 +681,15 @@ - (TestWebExtensionTab *)openNewTabAtIndex:(NSUInteger)index
     __weak TestWebExtensionTab *weakTab = newTab;
 
-    newTab.duplicate = ^(WKWebExtensionTabCreationOptions *options, void (^completionHandler)(TestWebExtensionTab *, NSError *)) {
+    newTab.duplicate = ^(WKWebExtensionTabConfiguration *options, void (^completionHandler)(TestWebExtensionTab *, NSError *)) {
```
```suggestion
    newTab.duplicate = ^(WKWebExtensionTabConfiguration *configuration, void (^completionHandler)(TestWebExtensionTab *, NSError *)) {
```

jeffmapple commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIAction.mm:
```diff
@@ -1282,11 +1282,11 @@
     };
 
-    manager.get().internalDelegate.openNewTab = ^(WKWebExtensionTabCreationOptions *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionTab>, NSError *)) {
+    manager.get().internalDelegate.openNewTab = ^(WKWebExtensionTabConfiguration *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionTab>, NSError *)) {
```
```suggestion
    manager.get().internalDelegate.openNewTab = ^(WKWebExtensionTabConfiguration *configuration, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionTab>, NSError *)) {
```

jeffmapple commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIAction.mm:
```diff
@@ -1324,7 +1324,7 @@
 #if PLATFORM(MAC)
     auto originalOpenNewWindow = manager.get().internalDelegate.openNewWindow;
-    manager.get().internalDelegate.openNewWindow = ^(WKWebExtensionWindowCreationOptions *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
+    manager.get().internalDelegate.openNewWindow = ^(WKWebExtensionWindowConfiguration *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
```
```suggestion
    manager.get().internalDelegate.openNewWindow = ^(WKWebExtensionWindowConfiguration *configuration, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionWindow>, NSError *)) {
```

jeffmapple commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIAction.mm:
```diff
@@ -1342,11 +1342,11 @@
 #else
     auto originalOpenNewTab = manager.get().internalDelegate.openNewTab;
-    manager.get().internalDelegate.openNewTab = ^(WKWebExtensionTabCreationOptions *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionTab>, NSError *)) {
+    manager.get().internalDelegate.openNewTab = ^(WKWebExtensionTabConfiguration *options, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionTab>, NSError *)) {
```
```suggestion
    manager.get().internalDelegate.openNewTab = ^(WKWebExtensionTabConfiguration *configuration, WKWebExtensionContext *context, void (^completionHandler)(id<WKWebExtensionTab>, NSError *)) {
```

xeenon committed:
Address additional API feedback for Web Extensions.
https://webkit.org/b/278040
rdar://problem/133783321

Reviewed by Jeff Miller.

Address various API review comments with tweaks to various property names, header docs, and some classes.

* Source/WebCore/en.lproj/Localizable.strings: Updated with update-webkit-localizable-strings.
* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKit/Modules/iOS_Private.modulemap:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtension.mm:
(-[WKWebExtension hasPersistentBackgroundContent]):
(-[WKWebExtension _hasServiceWorkerBackgroundContent]):
(-[WKWebExtension _hasModularBackgroundContent]):
(-[WKWebExtension backgroundContentIsPersistent]): Deleted.
(-[WKWebExtension _backgroundContentIsServiceWorker]): Deleted.
(-[WKWebExtension _backgroundContentUsesModules]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionAction.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommand.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionCommandPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionContext.mm:
(-[WKWebExtensionContext hasRequestedOptionalAccessToAllHosts]):
(-[WKWebExtensionContext setHasRequestedOptionalAccessToAllHosts:]):
(-[WKWebExtensionContext hasAccessToPrivateData]):
(-[WKWebExtensionContext setHasAccessToPrivateData:]):
(toImpl):
(-[WKWebExtensionContext requestedOptionalAccessToAllHosts]): Deleted.
(-[WKWebExtensionContext setRequestedOptionalAccessToAllHosts:]): Deleted.
(-[WKWebExtensionContext hasAccessInPrivateBrowsing]): Deleted.
(-[WKWebExtensionContext setHasAccessInPrivateBrowsing:]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionController.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerConfigurationPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionControllerDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionMessagePort.mm:
(-[WKWebExtensionMessagePort sendMessage:completionHandler:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTab.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTabConfiguration.h: Renamed from WKWebExtensionTabCreationOptions.h.
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTabConfiguration.mm: Renamed from WKWebExtensionTabCreationOptions.mm.
(-[WKWebExtensionTabConfiguration _init]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionTabConfigurationInternal.h: Renamed from WKWebExtensionTabCreationOptionsInternal.h.
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindow.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindowConfiguration.h: Renamed from WKWebExtensionWindowCreationOptions.h.
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindowConfiguration.mm: Renamed from WKWebExtensionWindowCreationOptions.mm.
(-[WKWebExtensionWindowConfiguration _init]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebExtensionWindowConfigurationInternal.h: Renamed from WKWebExtensionWindowCreationOptionsInternal.h.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionTab.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionTabCreationOptions.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionWindowCreationOptions.h:
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIExtensionCocoa.mm:
(WebKit::WebExtensionContext::extensionIsAllowedIncognitoAccess):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIRuntimeCocoa.mm:
(WebKit::WebExtensionContext::runtimeOpenOptionsPage):
(WebKit::WebExtensionContext::runtimeSendNativeMessage):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPITabsCocoa.mm:
(WebKit::WebExtensionContext::tabsCreate):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIWindowsCocoa.mm:
(WebKit::WebExtensionContext::windowsCreate):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm:
(WebKit::WebExtension::populateBackgroundPropertiesIfNeeded):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::setHasAccessToPrivateData):
(WebKit::WebExtensionContext::canOpenNewWindow const):
(WebKit::WebExtensionContext::userGesturePerformed):
(WebKit::WebExtensionContext::websiteDataStore const):
(WebKit::WebExtensionContext::setHasAccessInPrivateBrowsing): Deleted.
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionControllerCocoa.mm:
(WebKit::WebExtensionController::addUserContentController):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionTabCocoa.mm:
(WebKit::WebExtensionTab::WebExtensionTab):
(WebKit::WebExtensionTab::parameters const):
(WebKit::WebExtensionTab::changedParameters const):
(WebKit::WebExtensionTab::matches const):
(WebKit::WebExtensionTab::extensionHasAccess const):
(WebKit::WebExtensionTab::setReaderModeShowing):
(WebKit::WebExtensionTab::isReaderModeShowing const):
(WebKit::WebExtensionTab::isPlayingAudio const):
(WebKit::WebExtensionTab::detectWebpageLocale):
(WebKit::WebExtensionTab::captureVisibleWebpage):
(WebKit::WebExtensionTab::duplicate):
(WebKit::WebExtensionTab::shouldGrantPermissionsOnUserGesture const):
(WebKit::WebExtensionTab::isAudible const): Deleted.
(WebKit::WebExtensionTab::shouldGrantTabPermissionsOnUserGesture const): Deleted.
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionWindowCocoa.mm:
(WebKit::WebExtensionWindow::WebExtensionWindow):
(WebKit::WebExtensionWindow::extensionHasAccess const):
(WebKit::WebExtensionWindow::isPrivate const):
* Source/WebKit/UIProcess/Extensions/WebExtension.serialization.in:
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.cpp:
(WebKit::WebExtensionContext::userContentControllers const):
(WebKit::WebExtensionContext::pageListensForEvent const):
(WebKit::WebExtensionContext::processes const):
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
(WebKit::WebExtensionContext::hasAccessToPrivateData const):
(WebKit::WebExtensionContext::hasAccessInPrivateBrowsing const): Deleted.
* Source/WebKit/UIProcess/Extensions/WebExtensionTab.h:
(WebKit::WebExtensionTab::allChangedProperties):
* Source/WebKit/UIProcess/Extensions/WebExtensionWindow.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtension.mm:
(TestWebKitAPI::TEST(WKWebExtension, BackgroundParsing)):
(TestWebKitAPI::TEST(WKWebExtension, BackgroundPreferredEnvironmentParsing)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIAction.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIAction, NavigationOpensInNewTab)):
(TestWebKitAPI::TEST(WKWebExtensionAPIAction, WindowOpenOpensInNewWindow)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPICookies.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPICookies, GetAllCookieStoresWithPrivateAccess)):
(TestWebKitAPI::TEST(WKWebExtensionAPICookies, GetAllIncognitoWithPrivateAccess)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIDeclarativeNetRequest.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIDeclarativeNetRequest, BlockedLoadInPrivateBrowsingTest)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIExtension.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIExtension, InIncognitoContext)):
(TestWebKitAPI::TEST(WKWebExtensionAPIExtension, IsAllowedIncognitoAccess)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIRuntime.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIRuntime, ConnectNative)):
(TestWebKitAPI::TEST(WKWebExtensionAPIRuntime, ConnectNativeWithInvalidMessage)):
(TestWebKitAPI::TEST(WKWebExtensionAPIRuntime, ConnectNativeWithUndefinedMessage)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPITabs.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, Create)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, CreateWithSpecifiedOptions)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, CreateWithRelativeURL)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, Duplicate)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, DuplicateWithOptions)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, QueryWithPrivateAccess)):
(TestWebKitAPI::TEST(WKWebExtensionAPITabs, UpdatedEventWithPrivateAccess)):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIWindows.mm:
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, GetAll)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, GetAllWithPrivateAccess)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, Create)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, CreateWithRelativeURL)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, CreateWithRelativeURLs)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, CreateIncognitoWithoutPrivateAccess)):
(TestWebKitAPI::TEST(WKWebExtensionAPIWindows, CreateIncognitoWithPrivateAccess)):
* Tools/TestWebKitAPI/cocoa/TestWebExtensionsDelegate.h:
* Tools/TestWebKitAPI/cocoa/TestWebExtensionsDelegate.mm:
(-[TestWebExtensionsDelegate webExtensionController:openNewWindowUsingConfiguration:forExtensionContext:completionHandler:]):
(-[TestWebExtensionsDelegate webExtensionController:openNewTabUsingConfiguration:forExtensionContext:completionHandler:]):
(-[TestWebExtensionsDelegate webExtensionController:sendMessage:toApplicationWithIdentifier:forExtensionContext:replyHandler:]):
(-[TestWebExtensionsDelegate webExtensionController:openNewWindowWithOptions:forExtensionContext:completionHandler:]): Deleted.
(-[TestWebExtensionsDelegate webExtensionController:openNewTabWithOptions:forExtensionContext:completionHandler:]): Deleted.
(-[TestWebExtensionsDelegate webExtensionController:sendMessage:toApplicationIdentifier:forExtensionContext:replyHandler:]): Deleted.
* Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.h:
* Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.mm:
(-[TestWebExtensionManager initForExtension:extensionControllerConfiguration:]):
(-[TestWebExtensionTab isReaderModeActiveForWebExtensionContext:]):
(-[TestWebExtensionTab setReaderModeActive:forWebExtensionContext:completionHandler:]):
(-[TestWebExtensionTab duplicateForWebExtensionContext:withOptions:completionHandler:]):
(-[TestWebExtensionWindow openNewTabAtIndex:]):
(-[TestWebExtensionWindow isPrivateForWebExtensionContext:]):
(-[TestWebExtensionTab isReaderModeShowingForWebExtensionContext:]): Deleted.
(-[TestWebExtensionTab setReaderModeShowing:forWebExtensionContext:completionHandler:]): Deleted.
(-[TestWebExtensionWindow isUsingPrivateBrowsingForWebExtensionContext:]): Deleted.

Canonical link: https://commits.webkit.org/282252@main
